xds/internal/balancer/outlierdetection/balancer_test.go

```diff
@@ -270,20 +297,1551 @@ func (lbc *LBConfig) Equal(lbc2 *LBConfig) bool {
+	// in which the plurality goes from multiple to multiple should be a no-op,
+	// as the address continues to be ignored by outlier detection.
+	od.UpdateAddresses(scw1, []resolver.Address{
+		{
+			Addr: "address2",
+		},
+		{
+			Addr: "address3",
+		},
+	})
+	// Verify no downstream effects.
+	sCtx, cancel := context.WithTimeout(context.Background(), defaultTestShortTimeout)
+	defer cancel()
+	if _, err := scsCh.Receive(sCtx); err == nil {
+		t.Fatalf("no SubConn update should have been sent (no SubConn got ejected/unejected)")
+	}
+
+	// Update scw1 back to a single address, which is ejected. This should cause
+	// the SubConn to be re-ejected.
+	od.UpdateAddresses(scw1, []resolver.Address{
+		{
+			Addr: "address2",
+		},
+	})
+	// Verify scw1 got ejected (UpdateSubConnState called with TRANSIENT FAILURE).
+	gotSCWS, err = scsCh.Receive(ctx)
+	if err != nil {
+		t.Fatalf("Error waiting for Sub Conn update: %v", err)
+	}
+	if err = scwsEqual(gotSCWS.(subConnWithState), subConnWithState{
+		sc:    scw1,
+		state: balancer.SubConnState{ConnectivityState: connectivity.TransientFailure},
+	}); err != nil {
+		t.Fatalf("Error in Sub Conn update: %v", err)
+	}
+}
+
+func scwsEqual(gotSCWS subConnWithState, wantSCWS subConnWithState) error {
+	if !cmp.Equal(gotSCWS, wantSCWS, cmp.AllowUnexported(subConnWithState{}, testutils.TestSubConn{}, subConnWrapper{}, addressInfo{}), cmpopts.IgnoreFields(subConnWrapper{}, "scUpdateCh")) {
+		return fmt.Errorf("received SubConnState: %+v, want %+v", gotSCWS, wantSCWS)
+	}
 	return nil
 }
 
-func (errParseConfigBuilder) Name() string {
-	return "errParseConfigBalancer"
+type rrPicker struct {
+	scs  []balancer.SubConn
+	next int
+}
+
+func (rrp *rrPicker) Pick(balancer.PickInfo) (balancer.PickResult, error) {
+	sc := rrp.scs[rrp.next]
+	rrp.next = (rrp.next + 1) % len(rrp.scs)
+	return balancer.PickResult{SubConn: sc}, nil
 }
 
-func (errParseConfigBuilder) ParseConfig(c json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
-	return nil, errors.New("some error")
+// TestDurationOfInterval tests the configured interval timer. On the first
+// config received, the Outlier Detection balancer should configure the timer
+// with whatever is directly specified on the config. On subsequent configs
+// received, the Outlier Detection balancer should configure the timer with
+// whatever interval is configured minus the difference between the current time
+// and the previous start timestamp. For a no-op configuration, the timer should
+// not be configured at all.
+func (s) TestDurationOfInterval(t *testing.T) {
+	stub.Register(tcibname, stub.BalancerFuncs{})
+
+	od, _, cleanup := setup(t)
+	defer func(af func(d time.Duration, f func()) *time.Timer) {
+		cleanup()
+		afterFunc = af
+	}(afterFunc)
+
+	durationChan := testutils.NewChannel()
+	afterFunc = func(dur time.Duration, _ func()) *time.Timer {
+		durationChan.Send(dur)
+		return time.NewTimer(1<<63 - 1)
```

Comment:
What value is this? `math.MaxInt64`? why not use that instead?
And why do you return a timer with this value?